### PR TITLE
New upwind

### PR DESCRIPTION
**Pull request by shivin9**
Separating #145 and #130.

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Looks like it fails tests even after the packages finally got the update.

**Comment by shivin9** [comment]
Yeah, I don't know why the multidimensional operators are breaking...

File: src/derivative_operators/convolutions.jl [hunk 1]
```diff
@@ -1,153 +1,302 @@
 # mul! done by convolutions
-function LinearAlgebra.mul!(x_temp::AbstractVector{T}, A::DerivativeOperator, x::AbstractVector{T}; overwrite = true) where T<:Real
```
Comment (ChrisRackauckas) [inline src/derivative_operators/convolutions.jl]: don't remove this.

File: src/derivative_operators/convolutions.jl [hunk 2]
```diff
@@ -1,153 +1,302 @@
...
         xtempi = zero(T)
         cur_stencil = eltype(stencil) <: AbstractVector ? stencil[i-A.boundary_point_count] : stencil
         cur_coeff   = typeof(coeff)   <: AbstractVector ? coeff[i] : coeff isa Number ? coeff : true
         cur_stencil = use_winding(A) && cur_coeff < 0 ? reverse(cur_stencil) : cur_stencil
         for idx in 1:A.stencil_length
-            xtempi += cur_coeff * cur_stencil[idx] * x[i - mid + idx]
+            x_idx = use_winding(A) && cur_coeff < 0 ? x[i + mid - idx] : x[i - mid + idx]
+            xtempi += cur_coeff * cur_stencil[idx] * x_idx
         end
-        x_temp[i] = xtempi + !overwrite*x_temp[i]
```
Comment (ChrisRackauckas) [inline src/derivative_operators/convolutions.jl]: don't remove this.

**Comment by ChrisRackauckas** [comment]
For starters a merge error which removed all of the overwrite handling will definitely cause that issue.

**Pull request closed without merging**